common-android/src/main/java/com/amplitude/common/android/AndroidContextProvider.kt

```diff
@@ -351,7 +353,17 @@ class AndroidContextProvider(private val context: Context, locationListening: Bo
             if (!isLocationListening) {
                 return null
             }
-            if (!Utils.checkLocationPermissionAllowed(context)) {
+            if (!(
+                ContextCompat.checkSelfPermission(
+                        context,
+                        Manifest.permission.ACCESS_COARSE_LOCATION
```

Comment:
LGTM, so this permission request is moved to the outside app (customer app if they want this track)?